pytorch_lightning/core/hooks.py

```diff
@@ -547,29 +565,62 @@ def transfer_batch_to_device(self, batch, device)
                     batch = super().transfer_batch_to_device(data, device)
                 return batch
 
+        See Also:
+            - :func:`~pytorch_lightning.utilities.apply_func.move_data_to_device`
+            - :func:`~pytorch_lightning.utilities.apply_func.apply_to_collection`
+        """
+        device = device or self.device
+        return move_data_to_device(batch, device)
+
+    def on_before_batch_transfer(self, batch):
+        """
+        Override to alter or apply batch augmentations to your batch before it is transferred to the device.
+
+        Note:
+            This hook only runs on single GPU training and DDP (no data-parallel).
+
         Args:
-            batch: A batch of data that needs to be transferred to a new device.
-            device: The target device as defined in PyTorch.
+            batch: A batch of data that needs to be altered or augmented.
 
         Returns:
-            A reference to the data on the new device.
+            A batch of data
 
-        Note:
-            This hook should only transfer the data and not modify it, nor should it move the data to
-            any other device than the one passed in as argument (unless you know what you are doing).
+        Example::
+
+            def on_before_batch_transfer(self, batch):
+                batch['x'] = transforms(batch['x'])
+                return batch
+
+        See Also:
+            - :func:`~pytorch_lightning.core.on_after_batch_transfer`
+            - :func:`~pytorch_lightning.core.transfer_batch_to_device`
+        """
+        return batch
+
+    def on_after_batch_transfer(self, batch):
+        """
+        Override to alter or apply batch augmentations to your batch after it is transferred to the device.
 
         Note:
             This hook only runs on single GPU training and DDP (no data-parallel).
-            If you need multi-GPU support for your custom batch objects in ``dp`` or ``ddp2``,
-            you need to define your custom :class:`~torch.nn.parallel.DistributedDataParallel` or
-            override :meth:`~pytorch_lightning.core.lightning.LightningModule.configure_ddp`.
+
+        Args:
+            batch: A batch of data that needs to be altered or augmented.
+
+        Returns:
+            A batch of data
+
+        Example::
+
+            def on_after_batch_transfer(self, batch):
+                batch['x'] = gpu_transforms(batch['x'])
+                return batch
 
         See Also:
-            - :func:`~pytorch_lightning.utilities.apply_func.move_data_to_device`
-            - :func:`~pytorch_lightning.utilities.apply_func.apply_to_collection`
+            - :func:`~pytorch_lightning.core.on_before_batch_transfer`
+            - :func:`~pytorch_lightning.core.transfer_batch_to_device`
```

Comment:
I don't think this will work, it has to be pytorch_lightning.core.lightning.LightningModule.methodname

Reply:
done.

Reply:
![Screenshot from 2021-01-29 01-51-39](https://user-images.githubusercontent.com/30778939/106194470-e356a700-61d4-11eb-910f-6a824ecbe769.png)
still doesn't add the links

Reply:
maybe it has to be :meth: not :func: